src/xnatdcat/xnat_parser.py

```diff
@@ -156,3 +164,38 @@ def xnat_to_RDF(session: XNATSession, config: Dict) -> Graph:
         logger.warning("There were %d projects with invalid data for DCAT generation", failure_counter)
 
     return export_graph
+
+
+def xnat_private_project(project) -> bool:
+    """This function checks if an XNAT project is a private project
+
+    Parameters
+    ----------
+    project : XNAT ProjectData
+        The project of which the permission status needs to be investigated
+
+    Returns
+    -------
+    bool
+        Returns True if the project is private, False if it is protected or public
+
+    Raises
+    ------
+    XNATParserError
+        If the XNAT API returns an unknown value for accessibility
+    """
+    # The API URI is documented as part of the XNAT Project Attributes API at
+    # https://wiki.xnat.org/xnat-api/project-attributes-api
+    # As it is not exposed as part of the XNAT XSD, XNATPy does not generate a field for it
+
+    # The API documentation says it should be Title case, in practice XNAT returns lowercase
+    # Therefore I consider the case to be unreliable
+    accessibility = project.xnat_session.get(f'{project.uri}/accessibility').text.casefold()
+    if accessibility == "private".casefold():
```

Comment:
please consider the following to avoid spaghetti-long if-else statements 
```
is_private = False
allowed = ["private", "public", "protected"]
if accessibility.lower() not in allowed:
    raise XNATParserError(f"Unknown permissions of XNAT project: accessibility is '{accessibility}'")
if accessibility.lower() == "private":
    is_private = True
return is_private

```

Reply:
Will incorporate this, thanks for the suggestion! Makes it a bit more pythonic.